src/build.ml

```diff
@@ -189,15 +195,26 @@ let get_prog = function
     arr (fun _ -> Error f)
     >>> dyn_paths (arr (function Error _ -> [] | Ok x -> [x]))
 
-let prog_and_args ?(dir=Path.root) prog args =
-  Paths (Arg_spec.add_deps args Path.Set.empty)
-  >>>
-  (get_prog prog &&&
-   (arr (Arg_spec.expand ~dir args)
-    >>>
-    dyn_path_set (arr (fun (_args, deps) -> deps))
+let prog_and_args =
+  let make_globs globs =
+    List.map globs ~f:(fun { Arg_spec. exts ; dir } ->
+      paths_matching_ignore ~dir (fun p ->
```

Comment:
why ignore here?

Reply:
We don't the warnings if the directories don't exist. This is how our hand made globs currently behave.